### PR TITLE
adding file uploading lock while text conversion/classifiers/extractors are running

### PR DESCRIPTION
**Pull request by sweng66**

### CODE REVIEW

**Review by valearna — APPROVED** [review]
Looks good, thanks @sweng66 ! I added a minor comment on a function name and on the file types

File: agr_literature_service/api/crud/workflow_tag_crud.py [hunk 1]
```diff
@@ -690,3 +695,72 @@ def get_reference_workflow_tags_by_mod(
 
     tags = [dict(row) for row in rows]
     return tags
+
+
+def is_job_running_for_paper(db: Session, reference_curie: str, mod_abbreviation: str) -> Optional[str]:
```
[the remark this answers is not in the document]
Comment (sweng66) [reply agr_literature_service/api/crud/workflow_tag_crud.py:700]: Sounds great! Will fix that.

File: agr_literature_service/api/crud/referencefile_crud.py [hunk 2]
```diff
@@ -223,12 +224,16 @@ def file_upload(db: Session, metadata: dict, file: UploadFile, upload_if_already
                                 detail="The specified curie is not in the standard Alliance format and no cross "
                                        "references match the specified value.")
         metadata["reference_curie"] = ref_curie_res.curie
+    if metadata["mod_abbreviation"]:
+        job_type = is_job_running_for_paper(db, metadata["reference_curie"], metadata["mod_abbreviation"])
```
[the remark this answers is not in the document]
Comment (sweng66) [reply agr_literature_service/api/crud/referencefile_crud.py:228]: for all files, not just main PDF.

**Comment by sweng66** [comment]
@valearna Thanks for checking the PR. I have updated the new function name to 'is_file_upload_blocked'.

**Review by valearna — APPROVED** [review]

**Comment by valearna** [comment]
Thanks @sweng66 !

**Pull request merged**